CMakeLists.txt

```
@@ -141,17 +131,15 @@ if (WIN32)
   endif()
 endif()
 
-if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64")
-    target_compile_options(snmalloc_lib INTERFACE $<$<COMPILE_LANGUAGE:CXX>:-mcx16>)
-  elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "amd64")
-   target_compile_options(snmalloc_lib INTERFACE $<$<COMPILE_LANGUAGE:CXX>:-mcx16>)
-  elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "AMD64")
-   target_compile_options(snmalloc_lib INTERFACE $<$<COMPILE_LANGUAGE:CXX>:-mcx16>)
-  elseif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86")
-   target_compile_options(snmalloc_lib INTERFACE $<$<COMPILE_LANGUAGE:CXX>:-mcx16>)
-  endif()
-endif ()
+# detect support for cmpxchg16b; werror is needed to make sure mcx16 must be used by targets
+set(SNMALLOC_CMAKE_CXX_FLAGS_CACHE ${CMAKE_CXX_FLAGS})
+set(CMAKE_CXX_FLAGS "-Werror -Wextra -Wall")
+check_cxx_compiler_flag(-mcx16 SNMALLOC_COMPILER_SUPPORT_MCX16)
```

If you change this line
```suggestion
check_cxx_compiler_flag("-mcx16 -Werror -Wextra -Wall" SNMALLOC_COMPILER_SUPPORT_MCX16)
```
Then I think you can remove the need to cache and update `CMAKE_CXX_FLAGS`.